docs/source/Usage.md

```diff
@@ -204,6 +204,7 @@ The following is the list of the available analyzers you can run out-of-the-box.
 * `AbuseIPDB`: check if an ip was reported on [AbuseIPDB](https://www.abuseipdb.com/)
 * `Fortiguard`: scan an observable with the [Fortiguard URL Analyzer](https://www.fortiguard.com/webfilter)
 * `TorProject`: check if an IP is a Tor Exit Node
+* `OnionScan`: check a Tor domain for misconfigurations and info leaks.
```

Comment:
For the reasons mentioned beforehand, I would like to have OnionScan as an analyzer that is optional.

I think one good way to do that is to transform it as a docker analyzer (see https://intelowl.readthedocs.io/en/latest/Advanced-Usage.html#optional-analyzers). Tor and the onionscan binary should be integrated there. They are not necessary in the default image.

However, that would mean to add another optional container. That is not a good thing because, if a user wants to enable all the analyzers, they would already need to run too many containers.

So I think this is the reason why we should integrate OnionScan requirements in an already existing container. We just need to choose which one.
@eshaan7 thoughts?

Reply:
I think you're right. Having a separate container for all tor enabled tools along with the daemon would help keep the main app separate from the noise that happens to pass through tor.

Adding the daemon to a separate container would be an organized way to handle things, but hard on the CPU resources. Is it suitable if we make installer scripts for separate tools inside the container, and just pass them as command line args during install?

Reply:
> Having a separate container for all tor enabled tools along with the daemon would help keep the main app separate from the noise that happens to pass through tor.

Yep, a new service in `integrations/` for TOR enabled tools makes sense. Similar to how we have one for static analyzers, other for APK analyzers.

Reply:
I didn't know that onionscan requires an active tor daemon running otherwise I'd have suggested this from the very beginning. That's a mistake on my part for not properly reading through their README. :man_facepalming:

Reply:
As I mentioned, we have already a solution for this cases and I would not like to add other "tricks" to solve this issue.

The solution is: Docker-based analyzers (all the ones that are in the `integration` folder, the ones mentioned in the previous link I shared). 

There is also documentation about how to integrate them: https://intelowl.readthedocs.io/en/latest/Contribute.html#integrating-a-docker-based-analyzer

The concept is that you are separating "problematic" dependencies from the base image and add them in optional images. This is very useful because analyzers should not influence the base image because they are not mandatory to the overall application: most of the users would like to use just some of them and not everyone (they are more than 100).

So, as said, we already have some of this optional images and maybe they are already too many to be run together.
So, instead of creating a new docker-based integration, we could just add OnionScan dependencies and service in an already existing one (like `box-js`). The point is that this case someway is not related to the others. I mean, there are no other analyzers that needs Tor right now. So it would make sense to create a new docker-based analyzer for Tor-related services. If we do that (this is easier, you can do that), then we can try to reduce the amount of the overall number of docker-based analyzers by grouping some ot them together (`box-js` and `apk_analyzers` for instance)

Reply:
So for this new `tor_analyzers`, you'd need to do 4 things:

- Create new directory inside `integrations/` named `tor_analyzers`. (Refer to `apk_analyzers` directory, you need the same files and same directory structure). This also means onionscan binary should now be kept inside `integrations/tor_analyzers/` as it will be executed from this new container using the flask-shell2http package.
- Convert the `Onionscan` class to inherit from `DockerBasedAnalyzer` class. (Refer to other docker-based observable analyzers such as thug_url.py, etc.)
- Add entry for the `tor_analyzers` integration in the `start.py`.

Reply:
created an issue for this too: https://github.com/intelowlproject/IntelOwl/issues/842

Reply:
@CypherpunkSamurai so we discussed within ourselves and the final decision is to have a new container for tor enabled services (we will combine/compress other analyzers together for optimization)

So for this new `tor_analyzers`, you'd need to do 4 things:

- Create new directory inside `integrations/` named `tor_analyzers`. (Refer to `apk_analyzers` directory, you need the same files and same directory structure). This also means onionscan binary should now be kept inside `integrations/tor_analyzers/` as it will be executed from this new container using the flask-shell2http package.
- Modify the `Onionscan` class to inherit from `DockerBasedAnalyzer` class. (Refer to other docker-based observable analyzers such as thug_url.py, etc.)
- Add entry for the `tor_analyzers` integration in the `start.py` script.
- Edit `Advanced-Usage.md` add new row in the table under "Optional Analyzers".

Reply:
Ok Sir, On It! 😄